java/android/nnstreamer/src/main/jni/nnstreamer-native-singleshot.c

```diff
@@ -263,8 +263,10 @@ nns_native_single_invoke (JNIEnv * env, jobject thiz, jlong handle, jobject in)
   }
 
   /* do not free input/output tensors (direct access from object) */
-  g_free (in_data);
-  g_free (out_data);
+  if (in_data)
+    _ml_tensors_data_destroy_internal (in_data, FALSE);
+  if (out_data)
+    _ml_tensors_data_destroy_internal (out_data, FALSE);
```

Comment:
Same here. ```nns_parse_tensors_data``` returns error if data_h is null.

Reply:
updated, plz review the changes again.